core/field_angle.js

```diff
@@ -294,13 +294,15 @@ Blockly.FieldAngle.prototype.dropdownCreate_ = function() {
 };
 
 /**
- * Dispose of events belonging to the angle editor.
+ * Disposes of events and dom-references belonging to the angle editor.
  * @private
  */
 Blockly.FieldAngle.prototype.dropdownDispose_ = function() {
   Blockly.unbindEvent_(this.clickWrapper_);
   Blockly.unbindEvent_(this.moveWrapper1_);
   Blockly.unbindEvent_(this.moveWrapper2_);
+  this.gauge_ = null;
```

Comment:
This isn't necessary.

Reply:
I think in this case it is.

If the angle field was itself being destroyed, it wouldn't need to set properties to null, but since we are releasing something else we need to set the properties to null.

Otherwise the dropdown div elements will be kept around as detached nodes. Learned that the hard way haha.

You can verify by running the following test:
```
      function memoryStrike() {
        for (var i = 0; i < 1000; i++) {
          var field = Blockly.Xml.domToBlock(Blockly.Xml.textToDom(
            '<block type="test_fields_angle"/>'), workspace).getField('FIELDNAME');
          field.showEditor_();
          Blockly.hideChaff();
        }
      }
```
And then looking at the heap's detached nodes; with and without setting the properties to null.

I was going to send you some heap snapshots I took, but github doesn't want to process my zip files --.--